src/Model.jl

```diff
@@ -35,11 +37,19 @@ TODO docme
 If no smearing is specified the system will be assumed to be an insulator
 Occupation obtained as `f(ε) = smearing((ε-εF) / T)`
 """
-function Model(lattice::AbstractMatrix{T}, n_electrons; external=nothing,
+function Model(lattice::AbstractMatrix{T}; n_electrons=nothing, atoms=[], external=nothing,
                nonlocal=nothing, hartree=nothing, xc=nothing, temperature=0.0,
                smearing=nothing, spin_polarisation=:none) where {T <: Real}
     lattice = SMatrix{3, 3, T, 9}(lattice)
 
+    if n_electrons === nothing
+        # get it from the atom list
+        isempty(atoms) && @error "Either n_electrons or a non-empty atoms list should be provided"
```

Comment:
You should use `error()` here, since `@error` just prints, but does not abort execution (it does not throw).